src/test/java/picocli/ColorSchemeTest.java

```diff
@@ -149,6 +154,102 @@ public void testColorScheme_stringAnsiOn() {
         assertEquals("Hello #UNDERLINE ME# please.", colorScheme.string("Hello @|underline UNDERLINE ME|@ please."));
     }
 
+    @Test
```

Comment:
Can you add tests with Ansi.ON?
to verify that parts of the message have different colors and can be controlled by the ColorScheme?
Maybe test with a custom colorscheme also?

Reply:
Sure.

Reply:
Could you provide such an example of invoking with `Ansi.ON`?

Reply:
ok.
I added `ExecuteTest::testIssue1048CauseException`. This fails with current master but passes when your PR is applied.

Note the commented out `//System.setProperty("picocli.ansi", "true");` before `CommandLine::execute` is called. 

That is what I mean: let's add 2 more tests, both with `-Dpicocli.ansi=true`, one using the standard color scheme, and one with a customized color scheme. 

In the verification part of the test you would assert that the output contains the expected ANSI escape codes (similar to some tests you had previously): we want to verify that different styles are applied to the stack trace part and to the error part.

Reply:
I pushed one more commit to make it easier to add tests.

Reply:
Thanks! Now I understand. I'm working on it.